### ISSUE TITLE
Need progress indicator in several places

### ISSUE THREAD

**Issue by graybeal** [metadatacenter/cedar-project#238]
On staging, do a search. Then replace the search text and hit return. The original search text is displayed in the desktop header area ("Search Results For: 'John'"), even while the search is going on for the new search. The new search takes 5 seconds or more to execute, and you can't really tell when it's done (except that if the name is changed, you'll see it change in the title bar when the search is done). If you hit return again or quickly do another search, things sometimes get a little out of sync and you can't really tell what's going on.

So to sum up: Search results header should update immediately on entry of 'return' character in search field; message should appear when search is complete, or else you can't tell whether the search is complete when the results are empty; things can get fairly out of sync; and sometimes you get a server error.

**Comment by graybeal** [comment]
<img width="1306" alt="screen shot 2016-11-12 at 10 34 28 pm" src="https://cloud.githubusercontent.com/assets/1266896/20244107/8ef1f87c-a928-11e6-9eb0-cedf3d8f9acd.png">

**Comment by egyedia** [comment]
I think this will be solved if we add a "loading" animation. This should be done not only for the search, but all folder operations

**Comment by willrett** [comment]
I added a progress bar for search to branch issue-238
